pandas/tests/io/formats/test_to_html.py

```diff
@@ -883,3 +883,29 @@ def test_to_html_na_rep_and_float_format(na_rep):
   </tbody>
 </table>"""
     assert result == expected
+
+
+def test_to_html_float_format_object_col():
+    # GH#40024
+    df = DataFrame(data={"x": [1000.0, "test"]})
+    result = df.to_html(float_format=lambda x: f"{x:,.0f}")
+    expected = """<table border="1" class="dataframe">
+  <thead>
```

Comment:
the expected html should be in a file, but I see #36690 slipped through https://github.com/pandas-dev/pandas/pull/36690#discussion_r498745047 so OK to leave as follow-up

Reply:
ok, thanks that's good to know. Will make a follow-up fixing that